runtime/hashing/common/test/aws/smithy/kotlin/runtime/hashing/HashingSinkTest.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.hashing
+
+import aws.smithy.kotlin.runtime.io.SdkBuffer
+import aws.smithy.kotlin.runtime.io.SdkSink
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class HashingSinkTest {
+
+    @Test
+    fun testHashingSinkDigest() = run {
+        val byteArray = ByteArray(1024) { 0xf }
+        val buffer = SdkBuffer()
+        buffer.write(byteArray)
+
+        val hashingSink = HashingSink(Crc32c(), SdkSink.blackhole())
+
+        val expectedHash = Crc32c()
+
+        assertEquals(expectedHash.digest().decodeToString(), hashingSink.digest().decodeToString())
+        hashingSink.write(buffer, buffer.size)
+        expectedHash.update(byteArray)
+        assertEquals(expectedHash.digest().decodeToString(), hashingSink.digest().decodeToString())
+    }
+
+    @Test
+    fun testCrcSinkDigest() = run {
+        val byteArray = ByteArray(1024) { 0xf }
+        val buffer = SdkBuffer()
+        buffer.write(byteArray)
+
+        val hashingSink = HashingSink(Crc32(), SdkSink.blackhole())
+
+        val expectedHash = Crc32()
+
+        assertEquals(expectedHash.digest().decodeToString(), hashingSink.digest().decodeToString())
+        hashingSink.write(buffer, buffer.size)
+        expectedHash.update(byteArray)
+        assertEquals(expectedHash.digest().decodeToString(), hashingSink.digest().decodeToString())
+    }
```

Comment:
**Question**: Are these two tests identical?

Reply:
They weren't initially, but I refactored and they ended up being basically the same, just using a different hash function. I will remove the duplication

Reply:
Oh my eyes missed the `Crc32` vs `Crc32c`. That's great to test both, maybe just refactor the test itself into a helper method that accepts a hash implementation.